File: app/src/main/java/android/mahendra/attendancemanager/Daos/PeriodDao.java
package android.mahendra.attendancemanager.Daos;

import android.mahendra.attendancemanager.models.Period;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface PeriodDao {
    @Insert(onConflict = OnConflictStrategy.ABORT)
    void insert(Period period);

    @Update
    void update(Period period);

    @Delete
    void delete(Period period);

    @Query("Delete from period_table")
    void deleteAll();

    @Query("SELECT * FROM PERIOD_TABLE WHERE week_day=:weekDay")
    LiveData<List<Period>> getAllPeriodsOn(final int weekDay);

    @Query("SELECT * FROM PERIOD_TABLE WHERE subject_title=:subjectTitle")
    LiveData<List<Period>> getAllPeriodsOfSubject(String subjectTitle);
}

File: app/src/main/java/android/mahendra/attendancemanager/daos/PeriodDao.kt
package android.mahendra.attendancemanager.daos

import android.mahendra.attendancemanager.models.Period

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

import kotlin.collections.List

@Dao
interface PeriodDao {
    @Insert(onConflict = OnConflictStrategy.ABORT)
    suspend fun insert(period: Period)

    @Update
    suspend fun update(period: Period)

    @Delete
    suspend fun delete(period: Period)

    @Query("Delete from period_table")
    suspend fun deleteAll()

    @Query("SELECT * FROM period_table")
    fun getAllPeriods(): LiveData<List<Period>>

    @Query("SELECT * FROM PERIOD_TABLE WHERE week_day=:weekDay ORDER BY period_number")
    fun getAllPeriodsOn(weekDay: Int): LiveData<List<Period>>

    @Query("SELECT * FROM PERIOD_TABLE WHERE subject_title=:subjectTitle")
    fun getAllPeriodsOfSubject(subjectTitle: String): LiveData<List<Period>>

    @Query("DELETE FROM PERIOD_TABLE WHERE week_day=:weekDay AND period_number=:periodNumber")
    suspend fun deletePeriod(periodNumber: Int, weekDay: Int)
}
